src/translators/sql_translator.py

```diff
@@ -482,6 +483,8 @@ def _preprocess_values(self, e, original_attrs, col_names,
                 values.append(e[self.TIME_INDEX_NAME])
             elif cn == FIWARE_SERVICEPATH:
                 values.append(fiware_servicepath or '')
+            elif cn == 'instanceId':
+                values.append(str(uuid4()))
```

Comment:
shouldn;t this be a URI?

Reply:
I think uuid is fine, but this may be better checked agains NGSI-LD spec, eventually the "uri" could be build in combination with our APIs path

Reply:
yea, I seem to remember the spec suggest an URI but probably isn't mandatory. @pooja1pathak can you check for us, thanks!!

also if it's a uri, isn't it easier to just stick it in the db? e.g.

    uri = 'urn:....' + uuid

the reason why I think it should be in the Db rather than in the Python app is that if you want to migrate the data you won't have to go through the web app to rebuild the entity data. Ditto for querying the DB directly. Thoughts?

Reply:
@chicco785 ignore my comment, I read yours too quickly :-) I don't think you;ve ever said only the uuid should go in the DB :-) sorry

Reply:
from the spec (page 33):

```
"instanceId": a URI uniquely identifying a Property instance. System generated. Optional.
```

Reply:
Yes it is an Optional field. I have found value of ```instanceId``` for version 1.3 and 1.4 at [ngsi-ld](https://uri.etsi.org/ngsi-ld/v1/ngsi-ld-core-context-v1.4.jsonld) as:
```
    "instanceId": {
      "@id": "ngsi-ld:instanceId",
      "@type": "@id"
    }
```
I have searched for the example and found one for ScorpioBroker at [ScorpioBroker](https://githubmemory.com/repo/ScorpioBroker/ScorpioBroker/issues?cursor=Y3Vyc29yOnYyOpK5MjAyMC0xMi0xN1QxOToyNzozNiswODowMM4t475i&pagination=next&page=4) as:
```
"instanceId": "urn:ngsi-ld:021fae1a-4df7-4610-9c24-a726fd1a9569"
```
Below is the exact request:
```
GET /ngsi-ld/v1/temporal/entities/urn:ngsi-ld:Vehicle:A104?timerel=after&time=2020-01-01T00:00:00Z
```
```
{
    "id": "urn:ngsi-ld:Vehicle:A104",
    "type": "Vehicle",
    "speed": [
        {
            "type": "Property",
            "value": 90,
            "instanceId": "urn:ngsi-ld:021fae1a-4df7-4610-9c24-a726fd1a9569",
            "observedAt": "2020-11-07T18:00:00Z"
        },
        {
            "type": "Property",
            "value": 80,
            "instanceId": "urn:ngsi-ld:339616ac-5f42-4fc7-8c2b-58d7144b1457",
            "observedAt": "2020-11-06T17:00:00Z"
        }
    ],
    "@context": [
        "https://raw.githubusercontent.com/ScorpioBroker/ScorpioBroker/master/Examples/index.json",
        "https://uri.etsi.org/ngsi-ld/v1/ngsi-ld-core-context.jsonld"
    ]
}
```

Please let me know if I need to append ```urn:ngsi-ld:``` to the UUID generated. And can we use Python method or need to look other methods?

Reply:
@pooja1pathak i think then that it makes perfectly sense to save in the db `urn:ngsi-ld:UUID`

Reply:
not the expert, but i suppose `uuid()` method from python base is perfectly fine. @c0c0n3 ?

Reply:
@pooja1pathak 

> use Python method

yep, the way you have it (`uuid4`) is perfect, good stuff!

Reply:
I have updated the code. I am working on test cases now.